File: Source/FlashCards/Entities/Factoid.cs
namespace Tft.FlashCards.Entities
{
  using System;
  using System.Collections.Generic;
  using System.ComponentModel.DataAnnotations;
  using System.Data.Entity.ModelConfiguration;
  using System.Linq;
  using System.Web;

  public enum FactoidType
  {
    Text,
    Unicode,
    Audio,
    Image,
    Video
  }

  public class Factoid: Entity
  {
    public FactoidType FactoidType { get; set; }    
    public FactoidCategory FactoidCategory { get; set; }
    public FlashCard FlashCard { get; set; }
    public Guid FlashCardGuid { get; set; }
    public string Fact { get; set; }
  }
}

File: Source/ReactFlashCard/Views/FlashCard/Factoid.ts
enum FactoidType {
    Text,
    Unicode,
    Audio,
    Image,
    Video
}


interface Factoid {
    type: FactoidType;
    category: FactoidCategory;
    fact: string;
}

interface UnicodeFactoid extends Factoid {}

interface AudioFactoid extends Factoid {}

interface ImageFactoid extends Factoid {}

interface VideoFactoid extends Factoid { }

class FactoidEvaluator {
    constructor(aFactoid: Factoid) {

    }

    evaluate = (): boolean => {
        return false;
    };

}